Error getting the version of the configured credential helper: Process 'docker-credential-secretservice version' exited with status 1

<!-- Please fill out the following information to help us in analyzing the issue, but feel free to skip it if you don't have the information at hand or if it does not apply. Please remove everything which does not apply to the issue. -->

### Description
This error below occur when trying to `mvn install` the hello world sample project.
Looks like it's related to [Issue #896 ](https://github.com/fabric8io/docker-maven-plugin/issues/896).

```
$ mvn io.fabric8:docker-maven-plugin:0.28.0:run
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Building dmp-sample-helloworld 0.28.0
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- docker-maven-plugin:0.28.0:run (default-cli) @ dmp-sample-helloworld ---
[ERROR] DOCKER> Error occurred during container startup, shutting down...
[ERROR] DOCKER> Error getting the version of the configured credential helper [Process 'docker-credential-secretservice version' exited with status 1]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 1.102 s
[INFO] Finished at: 2019-01-29T10:20:38Z
[INFO] Final Memory: 17M/57M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.fabric8:docker-maven-plugin:0.28.0:run (default-cli) on project dmp-sample-helloworld: Error getting the version of the configured credential helper: Process 'docker-credential-secretservice version' exited with status 1 -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```
docker-credential-secretservice doesn't accept `version` argument and returns 1.
```
$ docker-credential-secretservice version
Unknown credential action `version`
$ echo Return code: $?
Return code: 1
```
### Info

* d-m-p version : 0.28.0

* Maven version (`mvn -v`) :
```
Apache Maven 3.5.2
Maven home: /usr/share/maven
Java version: 1.8.0_201, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-8-oracle/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.15.0-43-generic", arch: "amd64", family: "unix"
```
* Docker version : Docker version 18.09.1, build 4c52b90
* If it's a bug, how to reproduce :
```
$ git clone https://github.com/fabric8io/docker-maven-plugin.git
$ cd docker-maven-plugin/samples/helloworld/
$ mvn io.fabric8:docker-maven-plugin:0.28.0:run
```

--- a/doc/changelog.md
+++ b/doc/changelog.md
@@ -1,6 +1,6 @@
 # ChangeLog
 
-* 0.28.1
+* **0.28-SNAPSHOT**
   - Reintroduce minimal API-VERSION parameter in order to support docker versions below apiVersion 1.25
   - docs: Correct default image naming
   - Proxy settings are being ignored ([#1148](https://github.com/fabric8io/docker-maven-plugin/issues/1148))
@@ -13,6 +13,7 @@
   - Added 'autoRemove' option for running containers (#1179)
   - Added support for AWS EC2 instance roles when pushing to AWS ECR (#1186)
   - Add support for auto-pulling multiple base image for multi stage builds (#1057)
+  - Fix usage of credential helper that do not support 'version' command (#1159)
   
 * **0.28.0** (2018-12-13)
   - Update to JMockit 1.43

--- a/src/main/java/io/fabric8/maven/docker/util/AuthConfigFactory.java
+++ b/src/main/java/io/fabric8/maven/docker/util/AuthConfigFactory.java
@@ -230,7 +230,7 @@ public class AuthConfigFactory {
             log.debug("AuthConfig: credentials from ~/.m2/setting.xml");
             return ret;
         }
-        
+
         // check EC2 instance role if registry is ECR
         if (EcrExtendedAuth.isAwsRegistry(registry)) {
             try {
@@ -253,8 +253,8 @@ public class AuthConfigFactory {
     }
 
     // ===================================================================================================
-    
-    
+
+
     // if the local credentials don't contain user and password, use EC2 instance
     // role credentials
     private AuthConfig getAuthConfigFromEC2InstanceRole() throws IOException {
@@ -310,7 +310,7 @@ public class AuthConfigFactory {
             }
         }
     }
-        
+
     private AuthConfig getAuthConfigFromSystemProperties(LookupMode lookupMode) throws MojoExecutionException {
         Properties props = System.getProperties();
         String userKey = lookupMode.asSysProperty(AUTH_USERNAME);
@@ -423,9 +423,10 @@ public class AuthConfigFactory {
 
     private AuthConfig extractAuthConfigFromCredentialsHelper(String registryToLookup, String credConfig) throws MojoExecutionException {
         CredentialHelperClient credentialHelper = new CredentialHelperClient(log, credConfig);
-        log.debug("AuthConfig: credentials from credential helper/store %s version %s",
+        String version = credentialHelper.getVersion();
+        log.debug("AuthConfig: credentials from credential helper/store %s%s",
                   credentialHelper.getName(),
-                  credentialHelper.getVersion());
+                  version != null ? " version " + version : "");
         return credentialHelper.getAuthConfig(registryToLookup);
     }
 
@@ -523,7 +524,7 @@ public class AuthConfigFactory {
                           useOpenAuthModeProp, kubeConfigEnv != null ? kubeConfigEnv : "~/.kube/config"));
 
     }
-   
+
 
     private Server checkForServer(Server server, String id, String registry, String user) {
 

--- a/src/main/java/io/fabric8/maven/docker/util/CredentialHelperClient.java
+++ b/src/main/java/io/fabric8/maven/docker/util/CredentialHelperClient.java
@@ -28,11 +28,11 @@ public class CredentialHelperClient {
         return credentialHelperName;
     }
 
-    public String getVersion() throws MojoExecutionException {
+    public String getVersion() {
         try {
             return new VersionCommand().getVersion();
         } catch (IOException e) {
-            throw new MojoExecutionException("Error getting the version of the configured credential helper",e);
+            return null;
         }
     }
 
@@ -79,9 +79,6 @@ public class CredentialHelperClient {
 
         public String getVersion() throws IOException {
             execute();
-            if (version == null) {
-                log.verbose("The credentials helper \"%s\" didn't return a version string",CredentialHelperClient.this.credentialHelperName);
-            }
             return version;
         }
     }

--- a/src/test/java/io/fabric8/maven/docker/util/AuthConfigFactoryTest.java
+++ b/src/test/java/io/fabric8/maven/docker/util/AuthConfigFactoryTest.java
@@ -169,7 +169,7 @@ public class AuthConfigFactoryTest {
                 expectedException.expect(MojoExecutionException.class);
                 expectedException.expectCause(Matchers.<Throwable>allOf(
                         instanceOf(IOException.class),
-                        hasProperty("message",startsWith("Failed to start 'docker-credential-credHelper1-does-not-exist version'"))
+                        hasProperty("message",startsWith("Failed to start 'docker-credential-credHelper1-does-not-exist get'"))
                 ));
                 factory.createAuthConfig(isPush,false,null,settings,"roland","registry1");
             }
@@ -185,7 +185,7 @@ public class AuthConfigFactoryTest {
                 expectedException.expect(MojoExecutionException.class);
                 expectedException.expectCause(Matchers.<Throwable>allOf(
                         instanceOf(IOException.class),
-                        hasProperty("message",startsWith("Failed to start 'docker-credential-credsStore-does-not-exist version'"))
+                        hasProperty("message",startsWith("Failed to start 'docker-credential-credsStore-does-not-exist get'"))
                 ));
                 factory.createAuthConfig(isPush,false,null,settings,"roland",null);
             }
@@ -201,7 +201,7 @@ public class AuthConfigFactoryTest {
                 expectedException.expect(MojoExecutionException.class);
                 expectedException.expectCause(Matchers.<Throwable>allOf(
                         instanceOf(IOException.class),
-                        hasProperty("message",startsWith("Failed to start 'docker-credential-credsStore-does-not-exist version'"))
+                        hasProperty("message",startsWith("Failed to start 'docker-credential-credsStore-does-not-exist get'"))
                 ));
                 factory.createAuthConfig(isPush,false,null,settings,"roland","registry2");
             }

